tests/headers/template-fun-ty.hpp

```diff
@@ -0,0 +1,13 @@
+template <typename T>
+class Foo
+{
+  typedef T (FunctionPtr)();
```

Comment:
Would it also be good to test `using` type aliases with function pointers and template parameters?